### PR TITLE
Migrate essential packages

### PR DESCRIPTION
**Pull request by supreeth7**
This PR serves as a foundation change in the migration effort of Backplane CLI.

**Notable changes:-**
- Migrate the essentials utility and client packages.
- Globalize the `getBackplaneClient()` function to be more accessible. [Credits: @samanthajayasinghe]
- Generate mocks for these packages.
- Remove the sample `hello` command created during the initial stages of repo initialization.

Iterative changes/migration can be done over these introduced changes.

### CODE REVIEW

File: pkg/info/info.go [hunk 1]
```diff
@@ -3,4 +3,9 @@
 package info
 
 // Version of the backplane-cli
-const Version = "0.0.0"
+const (
+	Version = "0.0.0"
+
+	BACKPLANE_URL_ENV_NAME     = "BACKPLANE_URL"
+	BACKPLANE_CONFIG_FILE_NAME = ".backplane.json"
```
Comment (wanghaoran1988) [inline pkg/info/info.go:10]: BACKPLANE_CONFIG_FILE_PATH = "~/.backplane.json"

File: pkg/utils/util.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+	BackplaneApi "github.com/openshift/backplane-api/pkg/client"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+const (
+	ClustersPageSize             = 50
+	BackplaneApiUrlRegexp string = `(?mi)^https:\/\/api-backplane\.apps\.(.*)`
```
Comment (wanghaoran1988) [inline pkg/utils/util.go:23]: Let's remove this for now

File: pkg/utils/cluster.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	logger.Debugln("Finding target cluster from kube config")
+	cfg, err := clientcmd.BuildConfigFromFlags("", clientcmd.NewDefaultPathOptions().GetDefaultFilename())
+	if err != nil {
+		return BackplaneCluster{}, err
+	}
+
+	backplaneServerRegex := regexp.MustCompile(BackplaneApiUrlRegexp)
+	if !backplaneServerRegex.MatchString(cfg.Host) {
+		return BackplaneCluster{}, fmt.Errorf("the api server is not a backplane url, please make sure you login to the cluster using backplane")
+	}
```
Comment (wanghaoran1988) [inline pkg/utils/cluster.go:45]: let's remove this for now

File: pkg/utils/ocmWrapper.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+	// get backplane URL from BACKPLANE_URL env variables
+	bpURL, hasURL := os.LookupEnv(info.BACKPLANE_URL_ENV_NAME)
+	if hasURL {
+		if bpURL != "" {
+			return bpURL, nil
+		} else {
+			return "", fmt.Errorf("%s env variable is empty", info.BACKPLANE_URL_ENV_NAME)
+		}
+	} else {
```
Comment (samanthajayasinghe) [inline pkg/utils/ocmWrapper.go:148]: @supreeth7 pls, add [this unit test ](https://github.com/openshift/backplane-cli/blob/c7572c1f17b1a9f84146f673187fb5f42e1fc417/pkg/utils/ocmWrapper_test.go) for covering the BP_URL setting via env vars

**Comment by supreeth7** [comment]
@wanghaoran1988 ready for a second round of review.

/label tide/merge-method-squash

**Pull request merged**